app/code/Magento/InventoryCatalogApi/Api/BulkPartialInventoryTransferInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalogApi\Api;
+
+interface BulkPartialInventoryTransferInterface
+{
+    /**
+     * Run bulk partial inventory transfer for specified items.
+     *
+     * @param \Magento\InventoryCatalogApi\Api\Data\PartialInventoryTransferInterface $transfer
+     * @return \Magento\InventoryApi\Api\Data\SourceItemInterface[]
+     * @throws \Magento\Framework\Validation\ValidationException
+     */
+    public function execute($transfer): array;
```

Comment:
hey @bcerban ,

Sorry, probably some miscommunication took place when we were chatting yesterday regarding changing to this interface.

I believe it's better to make $originSource and $destinationSource as attributes to this business operation, but not a part of input data. What I mean is to make this interface look like:

```php

namespace Magento\InventoryCatalogApi\Api;

interface BulkPartialInventoryTransferInterface
{
    /**
     * Run bulk partial inventory transfer for specified items.
     *
     * @param string $originSourceCode
     * @param string $destinationSourceCode
     * @param \Magento\InventoryCatalogApi\Api\Data\PartialInventoryTransferItemInterface $itemsToMove
     * @return void
     * @throws \Magento\Framework\Validation\ValidationException
     */
    public function execute(string $originSourceCode, string $destinationSourceCode, array $itemsToMove): void;
}

--------------------------------------

namespace Magento\InventoryCatalogApi\Api\Data;

/**
 * @api
 */
interface PartialInventoryTransferItemInterface
{
    const SKU = 'sku';
    const QTY = 'qty';

    /**
     * @return string
     */
    public function getSku(): string;

    /**
     * @param string $sku
     */
    public function setSku(string $sku): void;

    /**
     * @return float
     */
    public function getQty(): float;

    /**
     * @param float $qty
     */
    public function setQty(float $qty): void;
}
```

In this case the semantic of BulkPartialInventoryTransferInterface gives us a clear idea that transfer could be made just from Origin to Desstination source, and we no need to have an excessive interface \Magento\InventoryCatalogApi\Api\Data\PartialInventoryTransferInterface which holds origin and destination sources as a part of it, 1 DTO for keeping PartialInventoryTransferItemInterface is perfectly fine.
Also, please pay attention that service returns `void` as most of the commands (services which suppose to modify data) in MSI and in CQRS in general.